server/server.go

```diff
@@ -835,14 +836,28 @@ func (server *Server) loadEntryPointConfig(entryPointName string, entryPoint *En
 	return rewrite, nil
 }
 
-func (server *Server) buildDefaultHTTPRouter() *mux.Router {
+func (server *Server) buildDefaultHTTPRouter(globalConfiguration GlobalConfiguration) *mux.Router {
 	router := mux.NewRouter()
-	router.NotFoundHandler = http.HandlerFunc(notFoundHandler)
+	router.NotFoundHandler = newNotFounderHandler(globalConfiguration)
 	router.StrictSlash(true)
 	router.SkipClean(true)
 	return router
 }
 
+func newNotFounderHandler(globalConfiguration GlobalConfiguration) http.Handler {
+	log.Debugf("Setting up not found handler to serve %d responses", globalConfiguration.NoRouteResponseCode)
+
+	responseCode := globalConfiguration.NoRouteResponseCode
+	if responseCode == 0 {
+		responseCode = http.StatusNotFound
+	}
+
```

Comment:
I implemented this sanity checking now, falling back to `http.StatusNotFound` and logging a warning. I realised that logging a warning leads to output when running the test directly, though it is not shown when running `make test-unit`. Therefore I guess its fine.